packages/core-data/package.json

```diff
@@ -25,6 +25,7 @@
 	"main": "build/index.js",
 	"module": "build-module/index.js",
 	"react-native": "src/index",
+	"types": "src/types/index.ts",
```

Comment:
I'm not sure we should publish these types quite yet.

If we do that though we'll need to add a CHANGELOG entry as it will be a public API change when the new version of the package is released. `typescript` will resolve a package's own declared types before looking for definitely typed definitions so publishing these types effectively squashes the [DT types](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/wordpress__core-data). In the past I believe we've determined this to qualify as a breaking change for the package.

Reply:
I'm fine with not publishing them yet, but `import { Comment } from '@wordpress/core-data'` doesn't work for me without this line. Perhaps a nested tsconfig.json would do the trick?

Reply:
No, indeed it will not work in other packages without it; that's a limitation we have to live with. Packages in Gutenberg consume each more-or-less in the same way third-party packages would use them so the APIs are restrictive.

Reply:
Alright! I wonder what to say in the README, though. I could just remove the Extending section entirely for now, or alternatively say that the feature is implemented but not enabled yet. Hm, I may go for the former and we'll restore it later on?

Reply:
I removed that section for now:

https://github.com/WordPress/gutenberg/pull/38666/commits/dde0888a5d80fe417f538d2d67b343841c716810

and moved it to a gist that I linked to in the description of this PR – we should be able to find it easily later on:

https://gist.github.com/adamziel/ec41157c10ac11286609a635adb35a0b

Reply:
Excellent, thanks!